[Feat]:유형선택뷰 기능 구현

s9hn opened the pull request:
1. 버튼 클릭 리스너 구현
2. intent 전달
3. selector 적용 및 해제 
에 대해서 코딩했습니다,,, 네,,

Commit:
[Feat]:유형선택뷰 기능 구현
1. 버튼 클릭 리스너 구현
2. intent 전달

hjh1161514 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -14,7 +14,7 @@
         <activity
             android:name=".feature.write.MediaSelectActivity"
-            android:exported="true" />
+            android:exported="false" />
```
`android:exported="true"`를 사용해야 다른 activity에서 열 수 있습니다! Main에서 MediaSelect뷰로 가기 위해서는 true로 설정해주세요😀

hjh1161514 commented on app/src/main/java/before/forget/feature/write/MediaSelectActivity.kt:
```diff
@@ -7,26 +7,54 @@ import before.forget.databinding.ActivityMediaSelectBinding
         super.onCreate(savedInstanceState)
         binding = ActivityMediaSelectBinding.inflate(layoutInflater)
 
         binding.clMediaBook.setOnClickListener {
```
전에 리뷰에서도 작성한 것처럼 `setOnClickListener` -> `initBtnClickListener()` 등 하나의 함수로 구현해서 onCreate 안에는 최대한 짧은 시간에 이해할 수 있도록 구현하는 것이 좋아요!

hjh1161514 commented on app/src/main/java/before/forget/feature/write/MediaSelectActivity.kt:
```diff
@@ -7,26 +7,54 @@ import before.forget.databinding.ActivityMediaSelectBinding
             binding.clMediaBook.isSelected = true
+            media = binding.tvMediaBook.text.toString()
+        }
+        binding.clMediaMovie.setOnClickListener {
```
binding이 계속 사용되는데, 이거는 `with(binding)`으로 작성하면 됩니다! 예를 들어서 `with(binding){ clMediaMovie.setOnClickListener()}`

hjh1161514 commented on app/src/main/java/before/forget/feature/write/MediaSelectActivity.kt:
```diff
@@ -7,26 +7,54 @@ import before.forget.databinding.ActivityMediaSelectBinding
+            media = binding.tvMediaBook.text.toString()
+        }
+        binding.clMediaMovie.setOnClickListener {
+            binding.clMediaMovie.isSelected = true
```
`isSelected = true`만 설정하면 다시 눌렀을 때 변화가 없어요. 다시 눌렀을 때 꺼지는 것도 구현해주세요~👍

hjh1161514 commented on app/src/main/res/values/strings.xml:
```diff
@@ -49,6 +49,30 @@
+    <string name="main_reportlabel">기록</string>
+    <string name="main_countlabel">N개</string>
+    <string name="main_viewallbtn">전체보기</string>
+
```
strings에서 뷰에 따른 string을 빠르게 확인하기 위해 줄바꿈이 필요한데 잘 해주셨어요! 근데 앞으로는 한 줄 씩 띄어지도록 확인해주세요!

Commit:
[Feat]:유형선택뷰 코드리뷰 반영
1. 버튼 클릭 리스너 함수 구현
2. 셀렉터 on/off(?)구현
3. 매니페스트 exported = true 로 설정

Commit:
[Feat]:유형선택뷰 코드리뷰 반영
1. 버튼 클릭 리스너 함수 구현
2. 셀렉터 on/off(?)구현
3. 매니페스트 exported = true 로 설정
+4. with(binding) 사용

jinsu4755 commented on app/src/main/java/before/forget/feature/main/MainAdapter.kt:
```diff
@@ -33,11 +33,10 @@ class MainAdapter : RecyclerView.Adapter<MainAdapter.MainViewHolder>() {
+                val mainToMyRecordIntent =
+                    Intent(itemView.context, MyRecordActivity::class.java)
+                mainToMyRecordIntent.putExtra("media", media)
+                ContextCompat.startActivity(itemView.context, mainToMyRecordIntent, null)
```
피드백 반영 굳
지금은 바쁘지만 나중엔 다른 피드백도 한번 해보자

jinsu4755 commented on app/src/main/java/before/forget/feature/write/MediaSelectActivity.kt:
```diff
@@ -7,26 +7,61 @@ import before.forget.databinding.ActivityMediaSelectBinding
+    private fun btnSetOnClickListener() {
+        with(binding) {
+            clMediaBook.setOnClickListener {
+                clMediaBook.isSelected = !clMediaBook.isSelected
+                media = tvMediaBook.text.toString()
```
요런 느낌의 코드가 반복되는데 그렇다면 해당 부분이 중복으로 발생하는 일이다 라고 해석할 수 있겠지?

그럼 하나의 일이니까 함수로 분리해줄 수 있지 않을까?
같은일을 반복해서 만들면 지금은 돌아가도 추후 변경이 있으면 모든 곳을 고쳐야해
예를들어 media에 view.text + "쓰기" 로 변하면 6번 고쳐야겠네

jinsu4755 reviewed (approved):
굳 일단 전체적으로 잘 하고 있슴다
워낙 일이 많으시니...더 많은 개선점은 우리 추후에 찾아보자...
